### ISSUE TITLE
[Bug]: aws_cloudfront_public_key without a newline permanently replaces itself

### ISSUE THREAD

**Issue by t0yv0** [hashicorp/terraform-provider-aws#40295]
### Terraform Core Version

1.9.8

### AWS Provider Version

5.77.0

### Affected Resource(s)

* aws_cloudfront_public_key

### Expected Behavior

`terraform apply` should succeed once and subsequently show no changes.

### Actual Behavior

`terraform apply` permanently shows changes.

### Relevant Error/Panic Output Snippet

```shell
N/A
```

### Terraform Configuration Files

# notice lack of newline at the end; if a newline is added the behavior is good
resource "aws_cloudfront_public_key" "pk" {
  encoded_key = "-----BEGIN PUBLIC KEY-----\nMIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAst2koIWbFf3mla5JWVVt\nBXssD2zJxeP+f744v/rWLkHGiSmnjLxChos+dIbU0KCfEFPhBeuRUFHItPcr1AD6\nZqnqntVPrOORJ92ySURLfePRiuoI3ZNo2ayhBlF7+3awsg8XUiU+67uJqcdhj/es\nq1bfj/TJlINOCriTvEyNaG8P9OwXj3pFVeUjuAk71gQNJRZWO8acnoRjnBzoCtHu\nI1HSlyrizNqeGlT31cmgbgRIBJPCG/MMHsBIzc0rD6Wilo+FsGxcGDwOGfF1oyjU\na4ejsjLZmyTPFi/76dncs2Y8nPykvVcD8NpqWeaksBHqpW754DBN+1o5/KpfUpi6\niwIDAQAB\n-----END PUBLIC KEY-----"
}

### Steps to Reproduce

Call ```terraform apply``` twice

### Debug Output

N/A

### Panic Output

N/A

### Important Factoids

It appears that de-facto the resources normalizes the key with a subsequent newline, however a relevant DiffSuppressFunc is missing to make this transformation benign.

### References

N/A

### Would you like to implement a fix?

No

**Comment by t0yv0** [comment]
Absolutely, good call. Also feel free to close as by/design. I missed that this was documented.

**Comment by justinretzolk** [comment]
I still agree that it's worth at least looking into adding a `DiffSupressFunc`, or otherwise handling this automatically via the provider. With that in mind, I'll leave this open, but update it a bit to indicate such. Thanks again for bringing this up!

**Comment by teohhanhui** [comment]
Has something changed in the meantime?

Following the advice [here](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/cloudfront_public_key):

> NOTE: When setting `encoded_key` value, there needs a newline at the end of string. Otherwise, multiple runs of terraform will want to recreate the `aws_cloudfront_public_key` resource.

I have this `storage.tf`:

```tf
locals {
  cloudfront_public_key_private = "${chomp(var.cloudfront_public_key_private)}\n"
}

resource "aws_cloudfront_public_key" "private" {
  comment     = "caribid-private"
  encoded_key = local.cloudfront_public_key_private
  name        = "caribid-private"
}
```

On the first `tofu plan`:

```
  # aws_cloudfront_public_key.private will be created
  + resource "aws_cloudfront_public_key" "private" {
      + caller_reference = (known after apply)
      + comment          = "caribid-private"
      + encoded_key      = <<-EOT
            -----BEGIN PUBLIC KEY-----
            MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEwTjQKzvqfLRB8FWq/7s6m7BUFgeV
            sGsov4uWEibNM7FkaP66jAmJkIMsBkN66xAXUC9bXuIoE6u3pKmDKoGMxg==
            -----END PUBLIC KEY-----
        EOT
      + etag             = (known after apply)
      + id               = (known after apply)
      + name             = "caribid-private"
      + name_prefix      = (known after apply)
    }
```

After `tofu apply`, on the next `tofu plan`:

```
  # aws_cloudfront_public_key.private must be replaced
-/+ resource "aws_cloudfront_public_key" "private" {
      ~ caller_reference = "terraform-20251113174856359100000001" -> (known after apply)
      ~ encoded_key      = <<-EOT # forces replacement
          - -----BEGIN PUBLIC KEY-----
          - MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEwTjQKzvqfLRB8FWq/7s6m7BUFgeV
          - sGsov4uWEibNM7FkaP66jAmJkIMsBkN66xAXUC9bXuIoE6u3pKmDKoGMxg==
          + -----BEGIN PUBLIC KEY-----
          + MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEwTjQKzvqfLRB8FWq/7s6m7BUFgeV
          + sGsov4uWEibNM7FkaP66jAmJkIMsBkN66xAXUC9bXuIoE6u3pKmDKoGMxg==
            -----END PUBLIC KEY-----
        EOT
      ~ etag             = "E23ZP02F085DFQ" -> (known after apply)
      ~ id               = "K2345VJRAYJGRJ" -> (known after apply)
        name             = "caribid-private"
      + name_prefix      = (known after apply)
        # (1 unchanged attribute hidden)
    }
```

UPDATE: Never mind. It seems like `\r\n` sneaked into the var somehow, so it's not related to this issue.
